src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -149,23 +162,30 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
         // check if already checked and passed the file
         if (!ordinaryChecks.isEmpty() || !commentChecks.isEmpty()) {
             final FileContents contents = getFileContents();
-            final DetailAST rootAST = JavaParser.parse(contents);
-            if (!ordinaryChecks.isEmpty()) {
-                walk(rootAST, contents, AstState.ORDINARY);
-            }
-            if (!commentChecks.isEmpty()) {
-                final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
-                walk(astWithComments, contents, AstState.WITH_COMMENTS);
-            }
-            if (filters.isEmpty()) {
-                addViolations(violations);
+            try {
+                final DetailAST rootAST = JavaParser.parse(contents);
+                if (!ordinaryChecks.isEmpty()) {
+                    walk(rootAST, contents, AstState.ORDINARY);
+                }
+                if (!commentChecks.isEmpty()) {
+                    final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
+                    walk(astWithComments, contents, AstState.WITH_COMMENTS);
+                }
+                if (filters.isEmpty()) {
+                    addViolations(violations);
+                }
+                else {
+                    final SortedSet<Violation> filteredViolations =
+                            getFilteredViolations(file.getAbsolutePath(), contents, rootAST);
+                    addViolations(filteredViolations);
+                }
+                violations.clear();
             }
-            else {
-                final SortedSet<Violation> filteredViolations =
-                    getFilteredViolations(file.getAbsolutePath(), contents, rootAST);
-                addViolations(filteredViolations);
+            catch (CheckstyleException ex) {
+                if (!skipFileOnJavaParseException) {
```

Comment:
This catch spans almost the entire method and has the possibility to catch more than just parsing exceptions. There is no check that we are catching parse exceptions. We could be catching much more.

This should be trimmed back to ensure it only catches parse exceptions.

There is issue https://github.com/checkstyle/checkstyle/issues/12803 to allow us to have more methods that throw checkstyle exceptions.

Reply:
@rnveach I agree, this is definitely not the best way to suppress expections. However, I was trying to only enclose `final DetailAST rootAST = JavaParser.parse(contents);` with try-catch. But, in that case, I need to use a `return` to block the following procedure, which will cause our CI to complain, since it is not allowed to have a return in a void method. Do you agree to suppress it?

Reply:
I do not agree to suppress yet. There should ways around this that follow our style rules.

Reply:
I can also add a local variable flag. When an exception occurs set the flag to true, and we guard the following procedure with this flag. If this flag is true, then we do not proceed the following procedure. What about this?

Reply:
Good catch, we should catch only parse exceptions, not CheckstyleException.

Reply:
We need to target with try catch only https://github.com/checkstyle/checkstyle/blob/86195e06404df68905c90232f71a1c88491f9ad8/src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java#L103

Reply:
done, work around by using a `skip` flag.

Reply:
@romani @rnveach This workaround using a `skip` flag seems bring us more suppressions with nullness. 
https://github.com/checkstyle/checkstyle/blob/e84dc9c47b2d593f52691dc19039b8c46fe2b261/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L169-L198
```
New surviving error(s) found:

  <checkerFrameworkError unstable="false">
    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java</fileName>
    <specifier>argument</specifier>
    <message>incompatible argument for parameter ast of TreeWalker.walk.</message>
    <lineContent>walk(rootAST, contents, AstState.ORDINARY);</lineContent>
    <details>
      found   : @Initialized @Nullable DetailAST
      required: @Initialized @NonNull DetailAST
    </details>
  </checkerFrameworkError>

  <checkerFrameworkError unstable="false">
    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java</fileName>
    <specifier>argument</specifier>
    <message>incompatible argument for parameter root of JavaParser.appendHiddenCommentNodes.</message>
    <lineContent>final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);</lineContent>
    <details>
      found   : @Initialized @Nullable DetailAST
      required: @Initialized @NonNull DetailAST
    </details>
  </checkerFrameworkError>

  <checkerFrameworkError unstable="false">
    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java</fileName>
    <specifier>argument</specifier>
    <message>incompatible argument for parameter rootAST of TreeWalker.getFilteredViolations.</message>
    <lineContent>getFilteredViolations(file.getAbsolutePath(), contents, rootAST);</lineContent>
    <details>
      found   : @Initialized @Nullable DetailAST
      required: @Initialized @NonNull DetailAST
    </details>
  </checkerFrameworkError>
```

I felt like this structure of code is too complicated. This is the core code of the project, it would be hard for later maintainers if we do more workarounds In this case, do you prefer to still work on this workaround (add a dummy initial value for rooAST instead of null), or prefer to suppress a single return statement, or there are any better solutions. Please share your thoughts. I am out of ideas.

Reply:
>it would be hard for later maintainers

If we want to have better code, please do this actively, right now this issue stuck in not done state, and users still having unfortunate decisions to disable checkstyle because it doesn't support modern language features.
You will not avoid catch of exception, so Checker will be not happy, and you will do just playing with creation of method that hide such exceptions handling, but conceptually all will stay the same.

If we start new method creation, you will stick in review more. My suggestion is pass review quicker, get feature out to users, and work in separate PR in beautiful code as much as we need, no rush. You can create new  PR, right now and collaborate on ideal form of code. 
There is no problem with mutation of variable, we do this in a lot of places, one day we will improve, but existence of this doesn't block users to receive bug fixes and new features.

Reply:
>My suggestion is pass review quicker, get feature out to users, and work in separate PR in beautiful code as much as we need, no rush.

Thanks for your advice. I will aim to finish this PR right now by suppressing the mutation test shown above. :)